sonar-html-plugin/src/main/java/org/sonar/plugins/html/checks/accessibility/ElementWithRoleShouldHaveRequiredPropertiesCheck.java

```diff
@@ -34,20 +37,23 @@ public void startElement(TagNode element) {
       return;
     }
 
-    var role = Aria.getRole(AriaRole.of(roleName));
-
-    if (role == null) {
-      return;
-    }
-
     var attributeNames = element.getAttributes().stream().map(Attribute::getName).toList();
-    var requiredProperties = role.getRequiredProperties();
-
-    for (var requiredProperty : requiredProperties) {
-      var requiredPropertyName = requiredProperty.toString();
-
-      if (!attributeNames.contains(requiredPropertyName)) {
-        createViolation(element, String.format("The attribute \"%s\" is required by the role \"%s\".", requiredPropertyName, roleName));
+    var roles = Arrays.stream(roleName.split("\\s+"))
+      .map(String::trim)
+      .map(AriaRole::of)
+      .filter(Objects::nonNull)
+      .map(Aria::getRole)
+      .filter(Objects::nonNull)
+      .toArray(RoleProperties[]::new);
+
+    for (var role : roles) {
+      var requiredProperties = role.getRequiredProperties();
+      for (var requiredProperty : requiredProperties) {
+        var requiredPropertyName = requiredProperty.toString();
+
+        if (!attributeNames.contains(requiredPropertyName)) {
+          createViolation(element, String.format("The attribute \"%s\" is required by the role \"%s\".", requiredPropertyName, role.getName()));
+        }
       }
```

Comment:
you gave up on streams here?

Reply:
kind of, found it easier to finish the traditional way